### ISSUE TITLE
how to update references in UITest ?

### ISSUE THREAD

**Issue by ZhangHanDong** [trailofbits/dylint#37]
I can run `cargo dev bless`to update references in Clippy , but i didn't find how to update references.

```rust

> cargo test

running 1 test
test [ui] ui/main.rs ... FAILED

failures:

failures:
    [ui] ui/main.rs

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.09s

test ui ... FAILED

failures:

---- ui stdout ----
"fill_me_in", "/Work/Projects/dylintt"
diff of stderr:

 error: function named `foo`
   --> $DIR/main.rs:9:5
    |
 LL |     pub fn foo(&self) {}
    |     ^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D foo-functions` implied by `-D warnings`
    = help: consider using a more meaningful name
 
 error: function named `foo`
   --> $DIR/main.rs:16:5
    |
 LL |     fn foo(&self) {}
    |     ^^^^^^^^^^^^^^^^
    |
    = help: consider using a more meaningful name
 
 error: function named `foo`
   --> $DIR/main.rs:22:1
    |
 LL | fn foo() {}
    | ^^^^^^^^^^^
    |
    = help: consider using a more meaningful name
 
 error: aborting due to 3 previous errors
+
+



The actual stderr differed from the expected stderr.
Actual stderr saved to /var/folders/q9/w05jvn_11tdblq6rzg5p_0t00000gn/T/main.stage-id.stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args main.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
```

**Comment by smoelius1** [comment]
Hi, @ZhangHanDong. Right now, the solution is to just update the `.stderr` file manually.

If I am reading your output correctly, I think you just need to add a couple of newlines at the end  file, as indicated by the two lines beginning with  `+`.

**Comment by smoelius** [comment]
Sorry for replying from the obscure account, @ZhangHanDong. I was traveling and didn't have access to my normal account.

Did the above suggestion work for you?

**Comment by ZhangHanDong** [comment]
@smoelius I added it manually but it hasn't worked yet, maybe I added it wrong and I gave up testing. I think you should add some instructions to the Readme or let it run automatically, it might work better.

**Comment by MinerSebas** [comment]
For me the UI Testing also doesn't work and it doesn't even tell me which Line is wrong:
```
failures:

---- ui stdout ----
diff of stderr:

 error: Unnecessary `With` Filter
   --> $DIR/bevy_lint.rs:10:34
    |
 LL | fn test_query1(_query: Query<&A, With<A>>) {}
    |                                  ^^^^^^^
    |
    = note: `-D bevy-lint` implied by `-D warnings`

 error: aborting due to previous error



The actual stderr differed from the expected stderr.
Actual stderr saved to C:\Users\scher\AppData\Local\Temp\bevy_lint.stage-id.stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args bevy_lint.rs`

error: 1 errors occurred comparing output.
status: exit code: 1
command: PATH=";C:\Rust\Bevy\bevy_dylint\target\debug\build\libgit2-sys-ab2169a8dd3f17b0\out\build;C:\Rust\Bevy\bevy_dylint\target\debug\build\libssh2-sys-8483915071c69b3b\out\build;C:\Rust\Bevy\bevy_dylint\target\debug\build\libz-sys-bba6d3c79f4f6fa3\out\lib;C:\Rust\Bevy\bevy_dylint\target\debug\deps;C:\Rust\Bevy\bevy_dylint\target\debug;C:\Users\scher\.rustup\toolchains\nightly-2021-06-03-x86_64-pc-windows-msvc\lib\rustlib\x86_64-pc-windows-msvc\lib;C:\Users\scher\.cargo\bin;C:\Users\scher\.rustup\toolchains\nightly-2021-06-03-x86_64-pc-windows-msvc\bin;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files\Python39\Scripts\;C:\Program Files\Python39\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\PuTTY\;C:\Program Files\Git\cmd;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\ProgramData\chocolatey\bin;C:\Program Files\dotnet\;C:\msys64\mingw64\bin;C:\Users\scher\.cargo\bin;C:\Users\scher\AppData\Local\Microsoft\WindowsApps;;C:\Users\scher\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\scher\.dotnet\tools" "C:\\Users\\scher\\.dylint_drivers\\nightly-2021-06-03-x86_64-pc-windows-msvc\\dylint-driver" "C:\\Users\\scher\\AppData\\Local\\Temp\\.tmpJMaBvs\\bevy_lint.rs" "-L" "C:\\Users\\scher\\AppData\\Local\\Temp\\" "--target=x86_64-pc-windows-msvc" "--error-format" "json" "-C" "prefer-dynamic" "-o" "C:\\Users\\scher\\AppData\\Local\\Temp\\bevy_lint.stage-id.exe" "-A" "unused" "--edition=2018" "-L" "dependency=C:\\Rust\\Bevy\\bevy_dylint\\target\\debug\\deps" "--extern" "bevy_ecs=C:\\Rust\\Bevy\\bevy_dylint\\target\\debug\\deps\\libbevy_ecs-ee997e710cdda8d5.rlib" "--extern" "clippy_utils=C:\\Rust\\Bevy\\bevy_dylint\\target\\debug\\deps\\libclippy_utils-bfe8420ca84aa50c.rlib" "--extern" "dylint_linting=C:\\Rust\\Bevy\\bevy_dylint\\target\\debug\\deps\\libdylint_linting-2330b2559771b6a7.rlib" "--extern" "dylint_testing=C:\\Rust\\Bevy\\bevy_dylint\\target\\debug\\deps\\libdylint_testing-aa909f8a2ba94b5e.rlib" "--extern" "if_chain=C:\\Rust\\Bevy\\bevy_dylint\\target\\debug\\deps\\libif_chain-2c5b2af41066c80b.rlib" "-L" "native=C:\\Rust\\Bevy\\bevy_dylint\\target\\debug\\build\\libgit2-sys-ab2169a8dd3f17b0\\out\\build" "-L" "native=C:\\Rust\\Bevy\\bevy_dylint\\target\\debug\\build\\libssh2-sys-8483915071c69b3b\\out\\build" "-L" "native=C:\\Rust\\Bevy\\bevy_dylint\\target\\debug\\build\\libz-sys-bba6d3c79f4f6fa3\\out\\lib" "-L" "native=C:\\Rust\\Bevy\\bevy_dylint\\target\\debug\\build\\libz-sys-bba6d3c79f4f6fa3\\out\\lib" "--emit=metadata" "-Dwarnings" "-Zui-testing" "-L" "C:\\Users\\scher\\AppData\\Local\\Temp\\bevy_lint.stage-id.aux"
stdout:
------------------------------------------

------------------------------------------
stderr:
{"message":"Unnecessary `With` Filter","code":{"code":"bevy_lint","explanation":null},"level":"error","spans":[{"file_name":"C:\\Users\\scher\\AppData\\Local\\Temp\\.tmpJMaBvs\\bevy_lint.rs","byte_start":134,"byte_end":141,"line_start":10,"line_end":10,"column_start":34,"column_end":41,"is_primary":true,"text":[{"text":"fn test_query1(_query: Query<&A, With<A>>) {}","highlight_start":3te_start":134,"byte_end":4,"highlight_end":41}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"`-D bevy-lint` implied by `-D warnings`","code":null,"abel":null,"suggested_replevel":"note","spans":[],"children":[],"rendered":null}],"rendered":"error: Unnecessary `With` Filter\n  --> C:\\Users\\scher\\AppData\\Local\\Temp\\.tmpJMaBvs\\bevy_lint.rs:10:34\n   |\nLL | fnnull}],"rendered":"error: test_query1(_query: Query<&A, With<A>>) {}\n   |                                  ^^^^^^^\n   |\n   = note: `-D bevy-lint` implied by `-D warnings`\n\n"}                                                ^^^^^^^\n   |\n  
{"message":"aborting due to previous error","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to previous error\n\n"}

------------------------------------------
```
